agent/pkg/api/test_helpers.go

```diff
@@ -72,6 +108,24 @@ func (m *MockBcloudClient) LinkerdMessages() []*pb.LinkerdMessage {
 	return messages
 }
 
+func (m *MockBcloudClient) ProxyDiagnosticMessages() []*pb.ProxyDiagnostic {
+	m.Lock()
+	defer m.Unlock()
+
+	messages := make([]*pb.ProxyDiagnostic, len(m.proxyDiagnosticMessages))
+	for i, m := range m.proxyDiagnosticMessages {
+		messages[i] = m
+	}
+
+	return messages
+}
```

Comment:
Just curious -- is it necessary to create a new slice here? Why not just return `m.proxyDiagnosticMessages`? Same goes for the other *Messages() functions above. It seems like they could all be simplified to:

```go
func (m *MockBcloudClient) ProxyDiagnosticMessages() []*pb.ProxyDiagnostic {
	m.Lock()
	defer m.Unlock()
	return m.proxyDiagnosticMessages
}
```

Reply:
I was under the impression that passing a pointer to an array like this would make it unsafe because one can modify this data structure in another go routing and trigger a data race. Cannot seem to cause that to happen 🤷

Reply:
Thanks for the details. That makes a lot of sense